plugins/modules/source_control/github/github_repo.py

```diff
@@ -0,0 +1,245 @@
+    type: bool
+    default: no
+    required: false
+  state:
+    description:
+    - Whether the repository should exist or not.
+    type: str
+    default: present
+    choices: [ absent, present ]
+    required: false
+  organization:
+    description:
+    - Organization for the repository.
+    - If not provided, the repository will be created in the current user profile.
+    type: str
+    required: false
+requirements:
+- PyGithub>=1.54
+notes:
+- For python3, PyGithub>=1.54 should be used.
+- "For python3.5, PyGithub==1.54 should be used. More info: https://pygithub.readthedocs.io/en/latest/changes.html#version-1-54-november-30-2020."
+- "For python2.7, PyGithub==1.45 should be used. More info: https://pygithub.readthedocs.io/en/latest/changes.html#version-1-45-december-29-2019."
+author:
+- Álvaro Torres Cogollo (@atorrescogollo)
+'''
+
+EXAMPLES = '''
+- name: Create a Github Repo
+  github_repo:
+    access_token: mytoken
+    organization: MyOrganization
+    name: myrepo
+    description: "Just for fun"
+    private: yes
+    state: present
+  register: result
+- name: Delete the repo
+  github_repo:
+    username: octocat
+    password: password
+    organization: MyOrganization
+    name: myrepo
+    state: absent
+  register: result
+'''
+
+RETURN = '''
+repo:
+  description: Repository information as JSON. See U(https://docs.github.com/en/rest/reference/repos#get-a-repository).
+  returned: success and I(state) is C(present)
+  type: dict
+'''
+
+
+import sys
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+import traceback
+
+GITHUB_IMP_ERR = None
+try:
+    from github import Github, GithubException
+    HAS_GITHUB_PACKAGE = True
+except Exception:
+    GITHUB_IMP_ERR = traceback.format_exc()
+    HAS_GITHUB_PACKAGE = False
+
+
+def authenticate(username=None, password=None, access_token=None):
+    if access_token:
+        return Github(base_url="https://api.github.com:443", login_or_token=access_token)
+    else:
+        return Github(base_url="https://api.github.com:443", login_or_token=username, password=password)
+
+
+def create_repo(gh, name, organization=None, private=False, description=''):
+    result = dict(
+        changed=False,
+        repo=None)
+    if organization:
+        target = gh.get_organization(organization)
+    else:
+        target = gh.get_user()
+
+    repo = None
+    try:
+        repo = target.get_repo(name=name)
+        result['repo'] = repo.raw_data
+
+        changes = {}
+        if repo.raw_data['private'] != private:
+            changes['private'] = private
+        if repo.raw_data['description'] != description:
+            changes['description'] = description
+
+        if changes:
+            repo.edit(**changes)
+
+        result['repo'].update({
+            'private': repo._private.value,
+            'description': repo._description.value
+        })
+    except GithubException as e:
+        if e.args[0] == 404:
```

Comment:
Instead of doing such checks, better catch `UnknownObjectException` (https://pygithub.readthedocs.io/en/latest/utilities.html#github.GithubException.UnknownObjectException).

Also, handling `UnknownObjectException` should be only done for the amount of code it's really necessary. If for some reason `repo.edit()` raises this exception, you're reacting wrong to it.